### DISCUSSION TITLE
Restoring from logged activity

### DISCUSSION THREAD

**Discussion by alantucker** [spatie/laravel-activitylog#960] [Q&A] [answered]
Hi

I need to offer the ability to restore logged activity e.g. if someone deletes a record, what's the best approach to do this? There will be instances when it's a single record or a record that has a relationship.

Thanks in advance.

Alan

**Comment by Gummibeer** [comment, marked as answer]
Hey,

this package isn't meant to be restoration capable. You would have to write and maintain your own logic.

The better solution would be event sourcing #594 which provides this out of the box.

**Reply by alantucker** [reply]
Great thanks very much, I will check it out.

Alan
